### PR TITLE
Build sweep mapping

### LINKED ISSUE #200: Handle data with re-acquired sweeps
If sweeps are re-acquired, then use acquisition with the latest starting time

### PR DESCRIPTION
**Pull request by sgratiy**
Resolve #200, #201

### CODE REVIEW

**Comment by sgratiy** [comment]
This is a redo of my previous PR which I am planning to delete.
This PR avoid merge conflict.
I wish I could have added tests for some of the new functions, but this requires either using real nwb files (for which we still have resolved the issue with where to store files) or making nwb test file data on-the-fly.

**Comment by sgratiy** [comment]
I have added a couple of commits related to the proper use of sweep numbers

File: ipfx/nwb_reader.py [hunk 1]
```diff
@@ -90,6 +90,9 @@ def read_sweep_from_source(source):
 
         raise ValueError("Could not find a source/sweep_number attribute/dataset.")
 
+    def get_starting_time(self, data_set_name):
+        NotImplementedError
```
Comment (njmei) [inline ipfx/nwb_reader.py]: Should this be `raise NotImplementedError`?

File: ipfx/nwb_reader.py [hunk 2]
```diff
@@ -410,6 +499,14 @@ def get_stim_code(self, sweep_name):
 
         return stim_code
 
+    def get_starting_time(self, data_set_name):
```
Comment (njmei) [inline ipfx/nwb_reader.py]: It looks like this function is duplicated twice (once in `class NwbXReader(NwbReader):` and once in `class NwbMiesReader(NwbReader):`). Maybe consider only defining it once in the `NwbReader` class?

Comment (t-b) [reply ipfx/nwb_reader.py]: It's not exactly duplicated, in one case it is an array, in the other a plain element. The function `get_scalar_string` (which should then also be renamed as it is not about strings anymore) can be used to make that distinction.

So the NwbReader class could look like:

```
    def get_starting_time(self, data_set_name):
        with h5py.File(self.nwb_file, 'r') as f:
            sweep_ts = f[self.acquisition_path][data_set_name]
            return get_scalar_string(sweep_ts["starting_time"].value)
```

**Comment by t-b** [comment]
@sgratiy 

> I wish I could have added tests for some of the new functions, but this requires either using real nwb files (for which we still have resolved the issue with where to store files) or making nwb test file data on-the-fly.

How about using my FTP for now? Just to keep us going? I'm mailing you the credentials via email.

**Comment by sgratiy** [comment]
@t-b 
We can continue using FTP for now.

**Comment by t-b** [comment]
Overall PR looks really good!

File: ipfx/nwb_reader.py (hunk 2 above)
Comment (sgratiy) [reply ipfx/nwb_reader.py]: @t-b Neat solution. I have adopted it. Thanks for a careful review and for the ftp info.

File: ipfx/nwb_reader.py (hunk 1 above)
Comment (sgratiy) [reply ipfx/nwb_reader.py]: @njmei Good catch, thanks

**Comment by sgratiy** [comment]
@t-b I have added a couple of tests using a nwb with reacquired sweeps and uploded it to the server. I assume this resolves the necessary changes.

**Pull request merged**